test/include/seqan3/test/sandboxed_path.hpp

```diff
@@ -0,0 +1,461 @@
+     * and additionally checks the invariant.
+     *
+     * ### Exceptions
+     *
+     * Basic exception guarantee.
+     */
+    template <typename path_t>
+    sandboxed_path & operator=(path_t const & path)
+    {
+        std::filesystem::path::operator=(path);
+        normalise();
+        checkInvariant();
+        return *this;
+    }
+
+    /*!\brief Replaces the path with a new path.
+     * \tparam path_t The type of the path.
+     *
+     * This works the same as std::filesystem::path::operator=
+     * and additionally checks the invariant.
+     *
+     * ### Exceptions
+     *
+     * Basic exception guarantee
+     */
+    template <typename path_t>
+    sandboxed_path & operator=(path_t && path)
+    {
+        std::filesystem::path::operator=(std::move(path));
+        normalise();
+        checkInvariant();
+        return *this;
+    }
+
+
+private:
+    /*!\brief Normalises the path.
+     *
+     * Normalisation means that the path is converted to an absolute path and
+     * is lexically normalised as described in std::filesystem::path.
+     *
+     * \sa https://en.cppreference.com/w/cpp/filesystem/path
+     */
+    void normalise()
+    {
+#if SEQAN3_WORKAROUND_GCC7_INCOMPLETE_FILESYSTEM
+        // convert into an absolute path
+        auto path = [this]() {
+            if (is_relative()) {
+                return (sandbox_path / *this).string();
+            } else {
+                return string();
+            }
+        }();
+        // Now we need to manually collapse any "." and ".."
+        size_t current_pos = path.find("/", 0); // find first "/" character
+        std::vector<std::string> path_parts;
+        while (current_pos < path.size()) {
+            auto end_pos = path.find("/", current_pos + 1);
+            auto word = path.substr(current_pos+1, end_pos-current_pos-1);
+            if (word == "." || word == "")
+            {
+                if (!path_parts.empty() and path_parts.back() != "")
+                {
+                    path_parts.emplace_back("");
+                }
+            }
+            else if (word == "..")
+            {
+                if (path_parts.empty())
+                {
+                    throw std::filesystem::filesystem_error("Path can not be normalised", *this,
+                                                             std::make_error_code(std::errc::invalid_argument));
+                }
+                path_parts.pop_back();
+            }
+            else
+            {
+                if (!path_parts.empty() and path_parts.back() == "")
+                {
+                    path_parts.back() = word;
+                }
+                else
+                {
+                    path_parts.emplace_back(word);
+                }
+            }
+            current_pos = end_pos;
+        }
+        std::string normalised_path;
+        for (auto const& p : path_parts) {
+            normalised_path += "/" + p;
+        }
```

Comment:
I don't want a view as a result, but an actuall string. It could be done with std::accumulate. But that seems quit anoying:
```
auto normalised_path = std::accumulate(begin(path_parts), end(path_parts), std::string{""}, [](auto prev, auto const& next)
{
    return prev + '/' + next;
});
```
Yes it is one line shorter, but one of the lines is way longer ;-) Not sure if it is what we want.